ste/xfer-remoteToLocal-file.go

```diff
@@ -352,11 +352,11 @@ func epilogueWithCleanupDownload(jptm IJobPartTransferMgr, dl downloader, active
 			}
 
 			//Rename back to original name. At this point, we're sure the file is completely
-			//downloaded and not corrupt. Infact, post this point we should log errors and
-			//fail the transfer.
-			if !strings.EqualFold(info.Destination, common.Dev_Null) {
-				err := os.Rename(info.getTempDownloadPath(), info.Destination)
-				if  err != nil {
+			//downloaded and not corrupt. Infact, post this point we should only log errors and
+			//not fail the transfer.
+			if err == nil && !strings.EqualFold(info.Destination, common.Dev_Null) {
+				renameErr := os.Rename(info.getTempDownloadPath(), info.Destination)
+				if  renameErr != nil {
 					jptm.LogError(info.Destination, fmt.Sprintf(
 								  "Failed to rename. File at %s", info.getTempDownloadPath()), err)
```

Comment:
@nakulkar-msft @mohsha-msft @zezha-msft  Should this line be referencing `renameErr` and not `err`? I'm hitting a "nil pointer dereference" error on this line when I call azcopy with --from-to=FileLocal.

I can open a separate issue if you'd like.

Reply:
Thanks for reaching out!

`err` should be nil here inside the if statement on line 357. 

Please do open a new issue, thank you!

Reply:
No problem! Thanks for the quick response. Issue created.

https://github.com/Azure/azure-storage-azcopy/issues/1421